openwebwork/webwork2
# Ww3 feature/dancer2

This is a pull request that changes the backend from a Dancer to a [Dancer2](https://metacpan.org/pod/Dancer2) web server without much functionality change. 

To test:

1. Following the steps laid out in the [installing webwork3](https://github.com/openwebwork/webwork2/wiki/Installing-WeBWorK3). 
2. If you are upgrading from an older webwork3, the configuration files `webwork3.conf` has changed in significant ways. Make sure you use a fresh copy of `webwork3.conf.dist`.
3. Once in a current course, click the WeBWorK3 link the instructor pages on webwork2.  If that is not present, check the configuration files again and look at [installing webwork3](https://github.com/openwebwork/webwork2/wiki/Installing-WeBWorK3). 
4. You may also login directly to a course from http://host/webwork3/courses/XXX/login .  Check this as both a student and an instructor.  You will get either a shell of a student view (to be added as a later pull request) or an instructor view (called the course manager).  
5. Try logging out/in as well as deleting cookies to try to ensure that security is high.  The cookies store the session ids.  
6. The general functionality of the course manager should be working, but there may be a few bugs.  However the goal of this PR is to get the overall switch to Dancer2 working.  

Note: I guess the current text editor I use is automatically stripping out a lot of white space causing much more file differences than there actually is. 

Lastly, PR #770 should be done before this and will shrink many of the commits/file changes.
